cli/src/commands/git/fetch.rs

```diff
@@ -72,13 +82,32 @@ pub fn cmd_git_fetch(
     args: &GitFetchArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let remotes = if args.all_remotes {
-        git::get_all_remote_names(workspace_command.repo().store())?
+    let remote_patterns = if args.all_remotes {
+        vec![StringPattern::everything()]
     } else if args.remotes.is_empty() {
         get_default_fetch_remotes(ui, &workspace_command)?
     } else {
         args.remotes.clone()
     };
+
+    let (mut remotes, remote_patterns): (Vec<_>, Vec<_>) =
+        remote_patterns.into_iter().partition_map(|p| {
+            if let Some(r) = p.as_exact() {
+                Either::Left(r.to_string())
+            } else {
+                Either::Right(p)
+            }
+        });
+
+    if !remote_patterns.is_empty() {
+        let all_remotes = git::get_all_remote_names(workspace_command.repo().store())?;
+        remotes.extend(
+            all_remotes
+                .into_iter()
+                .filter(|r| remote_patterns.iter().any(|p| p.matches(r))),
+        );
+    }
```

Comment:
In `jj bookmark` subcommands, we report patterns not matching anything as error, instead of handling exact patterns differently. (see `find_bookmarks_with()`.) I don't have preference over these, but for consistency, I think it's better to process remote name patterns in the same way.